code/game/objects/items/weapons/storage/RPED.dm

```diff
@@ -71,3 +71,63 @@
 		new /obj/item/weapon/stock_parts/capacitor/adv(src)
 	for(var/i in 1 to 8)
 		new /obj/item/weapon/stock_parts/console_screen(src)
+
+//Takes a tier 1 stock part path and a target rating
+//Returns a part object
+/proc/part_subtype(var/basepath, var/target)
+	for(var/path in subtypesof(basepath))
+		var/obj/item/weapon/stock_parts/SP = new path
+		if(SP.rating == target)
+			return SP
+		qdel(SP)
+
+/obj/item/weapon/storage/bag/gadgets/part_replacer/injector
+	name = "upgrade injector"
+	desc = "A single use upgrade injector. Just stab it into the side of a machine and it will dissolve away."
+	icon = 'icons/obj/syringe.dmi'
+	icon_state = "combat_hypo"
+	item_state = "hypo"
+	mech_flags = MECH_SCAN_FAIL
+	bluespace = TRUE
+	w_class = W_CLASS_SMALL
+	use_to_pickup = FALSE
+	var/base_rating = 2
+	var/list/part_types = list(/obj/item/weapon/stock_parts/manipulator,
+								/obj/item/weapon/stock_parts/matter_bin,
+								/obj/item/weapon/stock_parts/scanning_module,
+								/obj/item/weapon/stock_parts/capacitor,
+								/obj/item/weapon/stock_parts/micro_laser)
+
+/obj/item/weapon/storage/bag/gadgets/part_replacer/injector/super
+	name = "super upgrade injector"
+	icon_state = "combat_hypo_s"
+	base_rating = 3
+
+/obj/item/weapon/storage/bag/gadgets/part_replacer/injector/New()
+	..()
+	var/obj/item/weapon/stock_parts/SP
+	for(var/i in 1 to 3)
```

Comment:
Could make the `3` (max number of parts on the object) a variable on the pistol, in case someone creates stronger or weaker subtypes.